src/view/placeholder.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module engine/view/placeholder
+ */
+
+import extend from '@ckeditor/ckeditor5-utils/src/lib/lodash/extend';
+import EmitterMixin from '@ckeditor/ckeditor5-utils/src/emittermixin';
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+
+const CSS_CLASS = 'ck-placeholder';
+const DATA_ATTRIBUTE = 'data-placeholder';
```

Comment:
And I even think I know the answer. These kind of constants is useful to avoid magic strings and magic number. For instance:

```
const BLOCKS_SPACING = 15;

// ...

createBlock( BLOCK_SPACING );
```

is far more readable them:
```
createBlock( 15 );
```

But in this case, I'm not sure if these 2 are magic strings. I don't see anything hard to understand in:
```
element.setAttribute( 'data-placeholder', placeholderText );
```

I even like it more that the current version, because I immediately know what I need to look for in the browser's inspector in case of debugging.

So my proposal is to use this kind of constants only when the value or/and the context of its usage might be unclear and adding named const improve readability.